demo

```diff
@@ -62,16 +80,144 @@ if [[ "$?" -eq 1 ]]; then
     docker-compose pull ${compose_args}
 fi
 
-if [[ "$1" == "up" ]]; then
-    echo "Starting Mender demo environment.."
-fi
-
 # Pass this value on to the GUI container as an env variable
 export INTEGRATION_VERSION=$(git describe --tags --abbrev=0)
 
-exec docker-compose \
-     -f docker-compose.yml \
-     -f docker-compose.storage.minio.yml \
-     -f docker-compose.demo.yml \
-     $CLIENT \
-     "$@"
+if [[ "${!#}" != "up" ]]; then
```

Comment:
Done it like this, so that the command can take extra arguments, ie, -p `proj-name`. However, this is not currently consistent throughout the script. The previous method is `$1`. So I guess my question is: Which do you prefer?

Reply:
TBH I don't really understand how it works. Man page says "The entire command line typed so far". Won't this always fail to match then, since "!=" compares the whole string?

Reply:
The number of arguments is the last argument ;) tad hacky mby

Reply:
Suggestion: `echo "$@" | grep '\bup\b'`